### ISSUE TITLE
Multiple UptimeFlare instance in one account

### ISSUE THREAD

**Issue by alfari24** [lyc8503/UptimeFlare#49]
I want to create several of these status pages, but if I create another one, the old status page will be replaced with the new one.
Can you tell me how to prevent workers from being replaced?

**Comment by lyc8503** [comment]
Deploying multiple UptimeFlare instance in the same account will cause it to exceed the free quota limit for a single account, so it is not supported for now.

Maybe in the future I'll upgrade UptimeFlare to use D1 storage with much more free quota so I can support deploying multiple instances in one account.

Currently, as a temporary workaround, you can only create multiple Cloudflare accounts and then deploy UptimeFlare individually to each one. (This should not violate the current TOS for Cloudflare, which allow one person to hold multiple free accounts.)

**Comment by alfari24** [comment]
currently i have a paid plan for workers on cloudflare, 
so i have idea for make more status pages, for specific use. like personal use and my community used.

but when i trying used make another one page, they got replace on workers, so idk how to make diff worker. any idea ?

**Comment by lyc8503** [comment]
Due to some naming conflicts in the code that haven't been changed, you can't have two instances of UptimeFlare in the same account, even if you have a paid plan.

If you don't want to create multiple Cloudflare accounts, you can manually deploy UptimeFalre in wrangler with a different name so that the new deployment doesn't overwrite the old instance.

As for how to deploy manually, please refer to https://github.com/lyc8503/UptimeFlare/blob/main/.github/workflows/deploy.yml and https://github.com/lyc8503/UptimeFlare/blob/main/deploy.tf

**Comment by alfari24** [comment]
im still didnt get it the correct line need change.
after change few line
![image](https://github.com/user-attachments/assets/6551a77f-cb4e-438a-8cdd-571bdcfed6ef)

**Comment by lyc8503** [comment]
Seems you forgot to change `deploy.tf` Line 47, it should also be `new_project_name` in your case.

**Comment by alfari24** [comment]
This my **deploy.tf**

`terraform {
  required_providers {
    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "~> 4"
    }
  }
}

provider "cloudflare" {
  # read token from $CLOUDFLARE_API_TOKEN
}

variable "CLOUDFLARE_ACCOUNT_ID" {
  # read account id from $TF_VAR_CLOUDFLARE_ACCOUNT_ID
  type = string
}

resource "cloudflare_workers_kv_namespace" "uptimeflare_kv" {
  account_id = var.CLOUDFLARE_ACCOUNT_ID
  title      = "uptimeflare_kv"
}

resource "cloudflare_worker_script" "new_project_name" {
  account_id         = var.CLOUDFLARE_ACCOUNT_ID
  name               = "new_worker_name"
  content            = file("worker/dist/index.js")
  module             = true
  compatibility_date = "2023-11-08"

  kv_namespace_binding {
    name         = "UPTIMEFLARE_STATE"
    namespace_id = cloudflare_workers_kv_namespace.uptimeflare_kv.id
  }
}

resource "cloudflare_worker_cron_trigger" "uptimeflare_worker_cron" {
  account_id  = var.CLOUDFLARE_ACCOUNT_ID
  script_name = cloudflare_worker_script.new_project_name.name
  schedules = [
    "* * * * *", # every 1 minute, you can reduce the KV write by increase the worker settings of `kvWriteCooldownMinutes`
  ]
}

resource "cloudflare_pages_project" "new_project_name" {
  account_id        = var.CLOUDFLARE_ACCOUNT_ID
  name              = "new_project_name"
  production_branch = "main"

  deployment_configs {
    production {
      kv_namespaces = {
        UPTIMEFLARE_STATE = cloudflare_workers_kv_namespace.uptimeflare_kv.id
      }
      compatibility_date  = "2023-11-08"
      compatibility_flags = ["nodejs_compat"]
    }
  }
}`

**Comment by alfari24** [comment]
This my deploy.tf

```terraform {
  required_providers {
    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "~> 4"
    }
  }
}

provider "cloudflare" {
  # read token from $CLOUDFLARE_API_TOKEN
}

variable "CLOUDFLARE_ACCOUNT_ID" {
  # read account id from $TF_VAR_CLOUDFLARE_ACCOUNT_ID
  type = string
}

resource "cloudflare_workers_kv_namespace" "uptimeflare_kv" {
  account_id = var.CLOUDFLARE_ACCOUNT_ID
  title      = "uptimeflare_kv"
}

resource "cloudflare_worker_script" "new_project_name" {
  account_id         = var.CLOUDFLARE_ACCOUNT_ID
  name               = "new_worker_name"
  content            = file("worker/dist/index.js")
  module             = true
  compatibility_date = "2023-11-08"

  kv_namespace_binding {
    name         = "UPTIMEFLARE_STATE"
    namespace_id = cloudflare_workers_kv_namespace.uptimeflare_kv.id
  }
}

resource "cloudflare_worker_cron_trigger" "uptimeflare_worker_cron" {
  account_id  = var.CLOUDFLARE_ACCOUNT_ID
  script_name = cloudflare_worker_script.new_project_name.name
  schedules = [
    "* * * * *", # every 1 minute, you can reduce the KV write by increase the worker settings of `kvWriteCooldownMinutes`
  ]
}

resource "cloudflare_pages_project" "new_project_name" {
  account_id        = var.CLOUDFLARE_ACCOUNT_ID
  name              = "new_project_name"
  production_branch = "main"

  deployment_configs {
    production {
      kv_namespaces = {
        UPTIMEFLARE_STATE = cloudflare_workers_kv_namespace.uptimeflare_kv.id
      }
      compatibility_date  = "2023-11-08"
      compatibility_flags = ["nodejs_compat"]
    }
  }
}```

**Comment by lyc8503** [comment]
Try this change too

```diff
resource "cloudflare_workers_kv_namespace" "uptimeflare_kv" {
  account_id = var.CLOUDFLARE_ACCOUNT_ID
-  title      = "uptimeflare_kv"
+  title      = "new_uptimeflare_kv"
}
```

**Comment by alfari24** [comment]
still same.
![image](https://github.com/user-attachments/assets/882dc150-7a09-49b5-90bf-dc79bd0727ab)

**Comment by lyc8503** [comment]
See if there's anything wrong with `Deploy using Terraform` step.

**Comment by alfari24** [comment]
Every time I want to deploy I have to use a unique namespace, and it can't be the same every time I want to deploy. because it will conflict even though I deleted it in Cloudflare. this case is done

**Comment by lyc8503** [comment]
A note for future users: This is now being implemented in PR #97, after which you can deploy multiple instances in one account without changing source code.
